arrebol/worker/auth/auth.go

```diff
@@ -0,0 +1,78 @@
+package auth
+
+import (
+	"encoding/json"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/service/errors"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/key"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/token"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/whitelist"
+	"github.com/ufcg-lsd/arrebol-pb/crypto"
+)
+
+type Auth interface {
+	Validate(signature []byte, worker *worker.Worker) error
+	Authenticate( worker *worker.Worker) (*token.Token, error)
+	Authorize(token *token.Token) error
+}
+
+type JWTAuth struct {
+	workerKeyReader key.Reader
```

Comment:
why a reader and not the key itself?

Reply:
It's a reader of all workers' keys

Reply:
I got your idea, but we won't need to perform any operation with more than 1 worker key pair. So, I suggest you to remove the storage module, and keep some aux private functions from there in the Authenticator, like the getPubKey (once they're very simple, thanks for crypto.go). I think we're making some kind of over engineering by keeping a very simple module like storage.go. Agreed? Maybe that's not the best explanation, feel free to reach me out in case of any doubt

Reply:
Done.